### DISCUSSION TITLE
Add only specific parts of main prompts to AD prompts (dynamic prompting/wildcards fix)

### DISCUSSION THREAD

**Discussion by uorufu** [Bing-su/adetailer#553] [Ideas]
Adding specific parts of the main prompt to adetailer prompts through parsing.

for example if you have 
" [#1#2 prompt words here] [#1 extra] main prompt " 
as the main prompt.

and in adetailer prompts put say

" [#1] first tab" in first tab and " [#2] second tab" in second tab, 
first tab would add " prompt words here extra first tab" to the adetailer prompt 
and second tab only " prompt words here second tab" .

Something like this would would allow for use of wildcards and dynamic prompting to work in AD without having to resort of making code that relies on cooperation between extensions. It would simply work with these extensions and potential future extensions that are similar. Of course it assumed adetailer copies the main prompt after extensions such as wildcards and dynamic prompts have done their magic on the main prompt. 
I believe right now there is already something in place that places the entire prompt in adetailer by adding [prompt] to adetailer prompt, this would be a more expansive idea based on that.

You could also make it find specific lines from main prompt like "[line3]" or "[line3][line5]"  in adetailer prompt only adds the third line or the third and fifth line of the main prompt into the adetailer prompt respectively. 
This might be even better, as you dont need any special characters added into the main prompt.

**Comment by uorufu** [comment]
Ok, so I forked wildcards and made this.
[uorufu/stable-diffusion-webui-wildcards-adetailer](https://github.com/uorufu/stable-diffusion-webui-wildcards-adetailer)

This will do as I requested in OP. I'm no programmer and this was my first little project. Learned a lot though.

**Reply by sild928** [reply]
Great job - does exactly what I was looking for.

**Reply by kvsh88** [reply]
Could you please explain a bit as to how to use this properly with adetailer ?
